installer/win/BracketsConfigurator/BracketsConfigurator/BracketsConfigurator.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+using Microsoft.Deployment.WindowsInstaller;
+
+namespace BracketsConfigurator
+{
+    public class BracketsConfiguratorClass
+    {
+        [CustomAction]
+        public static ActionResult BracketsConfigurator(Session session)
+        {
+            session.Log("Begin BracketsConfigurator");
+
+            //Set Install Directory if available
+            string installRegistry = session["INSTALLDIRREGISTRY"];
+            string installDirValue = "";
+            if (!string.IsNullOrEmpty(installRegistry))
+            {
+                session.Log("Install Registry Value: " + installRegistry);
+                string[] path = installRegistry.Split('\\');
+                //Remove executable name
+                path = path.Take(path.Count() - 1).ToArray();
+                //Get path without executable
+                installDirValue = string.Join("\\", path) + "\\";
+                session["INSTALLDIR"] = installDirValue;
+                session.Log("Updating Install Dir To: " + installDirValue);
+            }
+            else
+            {
+                session.Log("No Registry Value for installation directory");
+            }
+
+            //Set AddContextMenu if required
+            string contextMenuRegistry = session["CONTEXTMENUREGISTRY"];
+            if (!string.IsNullOrEmpty(contextMenuRegistry))
+            {
+                session.Log("Context Menu Registry Value: " + contextMenuRegistry);
+                session["ADDCONTEXTMENU"] = "1";
+                session.Log("Setting Context Menu Option");
+            }
+            else
+            {
+                session["ADDCONTEXTMENU"] = "0";
+                session.Log("Not Adding to Context Menu");
+            }
+
+            //Set UpdatePath if required
+            string currentPathValue = session["CURRENTPATH"];
+            session.Log("Current Path Value: " + currentPathValue);
+            if (!string.IsNullOrEmpty(currentPathValue) && currentPathValue.Contains(installDirValue))
+            {
+                session["UPDATEPATH"] = "1";
+                session.Log("Updating Path for Brackets");
+            }
+            else
+            {
+                session["UPDATEPATH"] = "0";
```

Comment:
@gavankar In case we are unable to read the values from registry, I am seeing that we're defaulting to 0. Has this workflow been confirmed with upper management. Because the default values in case of a fresh launch I think is 1.

@ayushrathi15 can you confirm this as well?

Reply:
@mbhavi Adding Brackets to the system path is an optional feature in the installer.
This piece of code is used to figure out what an upgrading user choose the last time they ran the installer, not what the option should be for a new user.
If we see that Brackets is already on the path, updatepath is set to 1 and we continue to keep Brackets on the path. If brackets is not on the path then, we shouldn't add it now and updatepath is set to 0.

Reply:
@gavankar , I think my comment was misunderstood. What i meant was, if we are unable to read the registry, irrespective of whatever value it contains, what are we doing with the registry entries? Do we default them back to 0, or do we don't do anything? If we don't touch the registry entries, does the installer at the time of update, pick up the already existing entries, so that the user selected options are kept intact for the new version as well?
How are you handling this case here?

Reply:
@mbhavi, in this piece of code we are not directly reading entries from the registry. We are asking the msi installer for a copy of the entries that it has loaded as part of its execution, this is very unlikely to fail.